include/eosio/check.hpp

```diff
@@ -15,6 +33,57 @@ namespace eosio {
  *  @brief Defines wrappers over eosio_assert
  */
 
+
+struct eosio_error : std::exception {
+   explicit eosio_error(uint64_t code) {}
+};
+
+namespace detail {
+   [[noreturn]] inline void assert_or_throw(std::string_view msg) {
+#ifdef __eosio_cdt__
+         internal_use_do_not_use::eosio_assert_message(false, msg.data(), msg.size());
+#else
+         throw std::runtime_error(msg.data());
+#endif
+   }
+   [[noreturn]] inline void assert_or_throw(const char* msg) {
+#ifdef __eosio_cdt__
+         internal_use_do_not_use::eosio_assert(false, msg);
+#else
+         throw std::runtime_error(std::move(msg));
```

Comment:
Remove `std::move` as not used. Also `std::move` does nothing with `std::runtime_error` since it takes a `const string&`, but I'm fine leaving it in the `std::string&& msg` case below with the idea it might be added in the future.